src/main/java/org/opensearch/neuralsearch/search/collector/SimpleFieldCollector.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.search.collector;
+
+import java.io.IOException;
+import java.util.Objects;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.search.LeafCollector;
+import org.apache.lucene.search.Sort;
+import org.opensearch.neuralsearch.search.HitsThresholdChecker;
+
+/**
+ * SimpleFieldCollector collects the sorted results at the shard level for every individual query.
+ * It collects the list of TopFieldDocs.
+ */
+public final class SimpleFieldCollector extends HybridTopFieldDocSortCollector {
+    private final Sort sort;
+
+    public SimpleFieldCollector(int numHits, HitsThresholdChecker hitsThresholdChecker, Sort sort) {
+        super(numHits, hitsThresholdChecker, sort);
+        this.sort = sort;
+    }
+
+    @Override
+    public LeafCollector getLeafCollector(LeafReaderContext context) {
+        docBase = context.docBase;
+
+        return new HybridTopDocSortLeafCollector(sort, null) {
+            @Override
+            public void collect(int doc) throws IOException {
+                if (Objects.isNull(compoundQueryScorer)) {
+                    throw new IllegalArgumentException("scorers are null for all sub-queries in hybrid query");
+                }
+                float[] subScoresByQuery = compoundQueryScorer.hybridScores();
+                initializePriorityQueuesWithComparators(context, subScoresByQuery.length);
+                incrementTotalHitCount();
+                for (int i = 0; i < subScoresByQuery.length; i++) {
+                    float score = subScoresByQuery[i];
+                    // if score is 0.0 there is no hits for that sub-query
+                    if (score == 0) {
+                        continue;
+                    }
```

Comment:
This condition is making me shiver, as what I know is there can be 0 score value too at-least for k-NN with script score I have seen it.

Reply:
@martin-gaievski can you share your thoughts on how hybrid query is catering this case currently?

Reply:
for hybrid query we do not adding such hit to sub-query result -> it will not be in the final list of hits. If score = 0. is a valid case we may change the logic and add treat such hit similar to others. 
Change should be simple, score = 0 chosen as a stop value because we initialize array of hybrid scores with 0 in [scorer](https://github.com/opensearch-project/neural-search/blob/main/src/main/java/org/opensearch/neuralsearch/query/HybridQueryScorer.java#L183-L196). We can change it to something else and check for that new value instead of 0.

Reply:
Okay then lets open an issue separately to cater this change as it would require changes in current hybrid query workflow.  When the change in the current workflow would come we can cater the same in SimpleFieldCollector and Paging FieldCollector